hangman.py

```diff
@@ -0,0 +1,240 @@
+    than 30 times, program output random possible words.
+    '''
+
+    count_star = 0
+    guesses_point = 6
+    warnings_point = 3
+    letters_guessed = set()
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    print("You have", warnings_point, "warnings left")
+
+    while guesses_point > 0:
+        print("-------")
+        print("You have", guesses_point, "guesses left")
+        print("Available letters:", end=" ")
+        get_available_letters(letters_guessed)
+        letter = input("Please guess a letter:")
+        if request is False and letter == STAR:
+            check_star_30(count_star, secret_word, letters_guessed)
+            count_star += 1
+        else:
+            if letter == STAR:
+                warnings_point, guesses_point, count_star = classic_input(count_star, warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            elif letter not in string.ascii_letters or len(letter) != 1:
+                warnings_point, guesses_point = warnings("That is not a valid letter or you print more than one letter.", warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            elif letter in letters_guessed:
+                warnings_point, guesses_point = warnings("You have already guessed that letter:", warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            else:
+                letters_guessed, guesses_point = letter_in_word(secret_word, letter, letters_guessed, guesses_point, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed) == 0 or guesses_point <= 0:
+            break
+
+    #output result of game
+    if is_word_guessed(secret_word, letters_guessed) == 0:
+        final_points = guesses_point * len(set(secret_word))
+        print("------")
+        print("Congratulations, you won!")
+        print("Word:", secret_word)
+        print("Your total score for this game is:", final_points)
+    elif is_word_guessed(secret_word, letters_guessed) != 0 and guesses_point <= 0:
+        print("------")
+        print("Sorry, you ran out of guessed. The word was", secret_word)
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+
+    if len(my_word) == len(other_word):
+        for i in range(len(my_word)):
+            if (my_word[i] != '_' and my_word[i] != other_word[i]) or (my_word[i] == "_" and other_word[i] in my_word):
```

Comment:
I see you use symbol '_' not only once, and it has some meaning in that game. So maybe it makes sense to declare it as a constant?